app/scripts/modules/amazon/src/securityGroup/details/IPRangeRules.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Overridable, CollapsibleSection, LabeledValue, LabeledValueList } from '@spinnaker/core';
+
+export interface IIPRangeRulesProps {
+  ipRules: Array<IIPRangeRulesDetails>;
+}
+export interface IIPRangeRulesDetails {
+  rules: Array<IRangeRule>;
+  address: String;
+}
+
+export interface IRangeRule {
+  startPort: number;
+  endPort: number;
+  protocol: string;
+}
+
+@Overridable('aws.securityGroup.ip.rules')
+export class IPRangeRules extends React.Component<IIPRangeRulesProps> {
+  public render() {
+    const ipRules = this.props.ipRules || [];
+    const heading = 'IP Ranage Rules(' + ipRules.length + ')';
+    return (
+      <CollapsibleSection heading={heading}>
+        {ipRules.map(rule => (
+          <LabeledValueList className="horizontal-when-filters-collapsed">
+            <LabeledValue label="IP Range" value={rule.address} />
+            <LabeledValue
+              label="Port Ranges"
+              value={rule.rules.map(r =>
+                r.protocol === '-1' ? (
+                  <span>
+                    All ports and protocols
+                    {rule.rules.length > 1 ? (
+                      <div>
+                        <em>Additional port ranges are specified, but redundant:</em>
+                      </div>
+                    ) : null}
+                  </span>
+                ) : (
+                  <div>
+                    <span>
+                      {r.protocol + ':' + r.startPort} &rarr; {r.endPort}
+                    </span>
+                  </div>
```

Comment:
It could be worth making this a simple functional component for readability. 

```
<LabeledValue
   label="Port Ranges"
   value={rule.rules.map(r => 
       <PortRange rule={r} numRule={rule.rules.length}/>} 
/>
```

Where `<PortRange/>` just returns the logic in the highlighted lines. Totally up to you. This is definitely nit-picky and the main benefit is readability, and reusing the port range (not sure if that is needed though).